docs/concepts/configuration.md

```diff
@@ -51,8 +51,45 @@ As can be seen here, only build-time and deploy-time configuration parameter can
 
 ### Node.js Environment Variables
 
-When running the application in Angular Universal mode within a _Node.js_ environment, we can additionally access the process environment variables via _process.env._ This method provides a way to configure the application at deploy time, e.g., when using docker images.
-Configuration can then be consumed and passed to the client side via means of state transfer.
+When running the application in Angular Universal mode within a _Node.js_ environment, we can additionally access the process environment variables via _process.env_.
+This method provides a way to configure the application at deploy time, e.g., when using docker images.
+Configuration can then be consumed and passed to the client side via means of state transfer using Angulars [TransferState](https://angular.io/api/platform-browser/TransferState).
+
+To introduce a new `TransferState` key it should be added to the `src/app/core/configurations/state-keys.ts`.
+
+```typescript
+export const NEW_KEY = makeStateKey<string>('newKey');
+```
+
+The actual transfer is handled by the `src/app/app.server.module.ts` where the mapping of the environment variable is done.
+
+```typescript
+import { NEW_KEY } from 'ish-core/configurations/state-keys';
+
+export class AppServerModule {
+  constructor(transferState: TransferState) {
+    transferState.set(NEW_KEY, process.env.NEW_KEY);
+  }
+}
+```
```

Comment:
please link to official TransferState class instead of supplying a code snippet. Linking to `app.server.module.ts` would also be fine, so readers can look it up.

Reply:
I believe the 3 code snippets are helpful since they make it more obvious how the whole TransferState works together and what places need to be adapted. I think this would be less clear if I would remove one piece of it and leave this part to the reader to figure it out?

Reply:
I guess we should consider helping someone with this documentation that has not yet fully understood the underlying concept or would be able to get the usage just from reading https://angular.io/api/platform-browser/TransferState.

Reply:
For this, there is quite a variety of blog articles that cover this topic. And as we are not doing something entirely custom to the PWA, the developers should read about this on the web.